pubnub-entities/src/main/kotlin/com/pubnub/entities/User.kt

```diff
@@ -12,39 +23,104 @@ fun PubNub.getUsers(
     sort: Collection<PNSortKey> = listOf(),
     includeCount: Boolean = false,
     includeCustom: Boolean = false
-) = getAllUUIDMetadata(
-    limit = limit,
-    page = page,
-    filter = filter,
-    sort = sort,
-    includeCount = includeCount,
-    includeCustom = includeCustom
-)
+): ExtendedRemoteAction<PNUserArrayResult?> = firstDo(
+    getAllUUIDMetadata(
+        limit = limit,
+        page = page,
+        filter = filter,
+        sort = sort,
+        includeCount = includeCount,
+        includeCustom = includeCustom
+    )
+).then {
+    map(
+        it,
+        PNOperationType.UserOperation
+    ) { pnUuidMetadataArrayResult -> pnUuidMetadataArrayResult.toPNUserArrayResult() }
+
+}
 
 fun PubNub.getUser(
-    user: String? = null,
+    userId: String? = null,
     includeCustom: Boolean = false
-) = getUUIDMetadata(
-    uuid = user ?: configuration.uuid,
-    includeCustom = includeCustom
-)
+): ExtendedRemoteAction<PNUserResult?> = firstDo(
+    getUUIDMetadata(
+        uuid = userId ?: configuration.uuid,
+        includeCustom = includeCustom
+    )
+).then {
+    map(
+        it,
+        PNOperationType.UserOperation
+    ) { pnUuidMetadataResult -> pnUuidMetadataResult.toPNUserResult() }
+}
 
 fun PubNub.setUser(
-    user: String? = null,
+    userId: String? = null,
     name: String? = null,
     externalId: String? = null,
     profileUrl: String? = null,
     email: String? = null,
     custom: Any? = null,
     includeCustom: Boolean = false
-) = setUUIDMetadata(
-    uuid = user,
+): ExtendedRemoteAction<PNUserResult?> = firstDo(
+    setUUIDMetadata(
+    uuid = userId,
     name = name,
     externalId = externalId,
     profileUrl = profileUrl,
     email = email,
     custom = custom,
     includeCustom = includeCustom
 )
+).then {
+    map(
+        it,
+        PNOperationType.UserOperation
+    ) { pnUuidMetadataResult -> pnUuidMetadataResult.toPNUserResult() }
+}
+
+fun PubNub.deleteUser(
+    userId: String? = null
+): ExtendedRemoteAction<PNRemoveUserResult?> = firstDo(
+    removeUUIDMetadata(uuid = userId)
+).then {
+    map(
+        it,
+        PNOperationType.UserOperation
+    ){ pnRemoveMetadataResult ->  pnRemoveMetadataResult.toPNRemoveUserResult()}
+}
+
+internal fun PNUUIDMetadataArrayResult.toPNUserArrayResult(): PNUserArrayResult? {
```

Comment:
IMO that approach will make code cleaner but toPNUserArrayResult() method return object of type "PNUserArrayResult". This type is defied in optional pubnub-entities module that might be not included and in case it's included cyclic dependency occurs between  "pubnub-entities" and it's parent "pubnub-kotlin".